irmaclient/session.go

```diff
@@ -488,20 +489,26 @@ func (session *session) doSession(proceed bool, choice *irma.DisclosureChoice) {
 		session.finish(false)
 	} else {
 		var err error
-		session.builders, session.attrIndices, session.issuerProofNonce, err = session.getBuilders()
-		if err != nil {
-			session.fail(&irma.SessionError{ErrorType: irma.ErrorCrypto, Err: err})
-		}
-		startKeyshareSession(
+		keyshareSession, auth := newKeyshareSession(
 			session,
 			session.client,
 			session.Handler,
-			session.builders,
 			session.request,
 			session.implicitDisclosure,
-			session.issuerProofNonce,
-			session.timestamp,
+			session.Version,
 		)
+		if !auth {
+			// newKeyshareSession() calls session.fail() in case of failure, no need to do that here
+			return
+		}
+		session.builders, session.attrIndices, session.issuerProofNonce, err = session.getBuilders(keyshareSession)
+		if err != nil {
+			session.fail(&irma.SessionError{ErrorType: irma.ErrorCrypto, Err: err})
+		}
```

Comment:
Shoudn't we return after the `session.fail`?

Reply:
Yes! Thanks, added.